### PR TITLE
FINERACT-1022 - Update to org.ehcache

### PR DESCRIPTION
**Pull request by xurror**
sues.apache.org/jira/browse/FINERACT-1022

### CODE REVIEW

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -94,7 +94,7 @@ dependencyManagement {
         dependency 'org.flywaydb:flyway-core:6.4.4'
         dependency 'org.quartz-scheduler:quartz:2.3.2'
         dependency 'com.amazonaws:aws-java-sdk-s3:1.11.800'
-        dependency 'net.sf.ehcache:ehcache:2.10.6' // TODO FINERACT-1018
+        dependency 'org.ehcache:ehcache:3.8.1' // TODO FINERACT-1018
```
Comment (vorburger) [inline fineract-provider/build.gradle]: rm TODO ?

File: fineract-provider/build.gradle [hunk 2]
```diff
@@ -110,6 +110,7 @@ dependencyManagement {
         dependency 'io.github.classgraph:classgraph:4.8.85'
         dependency 'org.awaitility:awaitility:4.0.3'
         dependency 'com.github.spotbugs:spotbugs-annotations:4.0.4'
+        dependency 'javax.cache:cache-api:1.1.1'
```
Comment (vorburger) [inline fineract-provider/build.gradle:113]: Isn't that a transitive dependency of ehCache already?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/PlatformCacheConfiguration.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
  */
 package org.apache.fineract.infrastructure.cache;
 
+// import javax.cache.CacheManager;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/PlatformCacheConfiguration.java]: Don't add new commented out imports, but clean up

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/service/RuntimeDelegatingCacheManager.java [hunk 4]
```diff
@@ -120,6 +123,11 @@ public Cache getCache(final String name) {
     }
 
     private void clearEhCache() {
-        this.ehcacheCacheManager.getCacheManager().clearAll();
+        // this.jcacheCacheManager.getCacheManager().removeAll();
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/service/RuntimeDelegatingCacheManager.java]: rm? Is there no removeAll() on the new standard API?

**Review by vorburger — CHANGES_REQUESTED** [review]
Nice!! Thanks for this. Just some minor feedback.. in general, never comment out, but always just cleanly remove. The history is always there in Git anyway..

File: fineract-provider/build.gradle (hunk 1 above)
Comment (xurror) [reply fineract-provider/build.gradle]: 😂😂😂 Completely forgot.

File: fineract-provider/build.gradle (hunk 2 above)
Comment (xurror) [reply fineract-provider/build.gradle:113]: I wish it were but unfortunately it isn't. I tried without and it didn't work.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/service/RuntimeDelegatingCacheManager.java (hunk 4 above)
Comment (xurror) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/cache/service/RuntimeDelegatingCacheManager.java]: Nopes. Unfortunately there isn't. Just a destroy(still not destroyAll) method which I don't think is what we want to do.

**Review by vorburger — APPROVED** [review]
LGTM

**Comment by vorburger** [comment]
FINERACT-1022 says this may be broken, and we thus may revert this again.

**Pull request merged**